src/unwind_dwfl.cc

```diff
@@ -248,66 +247,77 @@ DDRes add_runtime_symbol_frame(UnwindState *us, const Dso &dso, ElfAddress_t pc,
 
   return add_frame(symbol_idx, map_idx, pc, us);
 }
+
+bool try_attach_module(UnwindState *us, const DsoHdr::DsoMap &map,
+                       bool &backpopulate_attempt) {
+  bool success = false;
+  for (auto it = map.cbegin(); it != map.cend(); ++it) {
+    const Dso &dso = it->second;
+    if (!dso.is_executable() || dso._type != DsoType::kStandard) {
+      continue;
+    }
+    const FileInfoId_t file_info_id = us->dso_hdr.get_or_insert_file_info(dso);
+    if (file_info_id <= k_file_info_error) {
+      LG_DBG("Unable to find file for DSO %s", dso.to_string().c_str());
+      continue;
+    }
+    const FileInfoValue &file_info_value =
+        us->dso_hdr.get_file_info_value(file_info_id);
+    DDProfMod *ddprof_mod = nullptr;
+    auto res = us->_dwfl_wrapper->register_mod(us->current_ip,
+                                               DsoHdr::get_elf_range(map, it),
+                                               file_info_value, &ddprof_mod);
+    if (IsDDResOK(res)) {
+      success = true;
+      break;
+    }
+    if (!backpopulate_attempt) {
+      file_info_value.reset_errored();
+    }
+  }
+  return success;
+}
 } // namespace
 
 DDRes unwind_init_dwfl(UnwindState *us) {
-  // Create or get the dwfl object associated to cache
   us->_dwfl_wrapper = us->dwfl_hdr.get_or_insert(us->pid);
   if (!us->_dwfl_wrapper) {
     return ddres_warn(DD_WHAT_UW_MAX_PIDS);
   }
-  if (!us->_dwfl_wrapper->_attached) {
-    // we need to add at least one module to figure out the architecture (to
-    // create the unwinding backend)
-
-    DsoHdr::DsoMap const &map = us->dso_hdr.get_pid_mapping(us->pid)._map;
-    if (map.empty()) {
-      int nb_elts;
-      us->dso_hdr.pid_backpopulate(us->pid, nb_elts);
-    }
+  if (us->_dwfl_wrapper->_attached) {
+    return {};
+  }
+  bool backpopulate_attempt = false;
```

Comment:
I would rename `backpopulate_attempt` to `backpopulate_done` to make it clearer.